src/main/webapp/app/exercises/quiz/manage/quiz-exercise.component.ts

```diff
@@ -158,12 +158,13 @@ export class QuizExerciseComponent extends ExerciseComponent {
      * @param quizExerciseId the quiz exercise id to end
      */
     endQuiz(quizExerciseId: number) {
-        this.quizExerciseService.end(quizExerciseId).subscribe({
+        return this.quizExerciseService.end(quizExerciseId).subscribe({
             next: (res: HttpResponse<QuizExercise>) => {
                 this.handleNewQuizExercise(res.body!);
+                this.dialogErrorSource.next('');
             },
-            error: (res: HttpErrorResponse) => {
-                this.onError(res);
+            error: (error: HttpErrorResponse) => {
+                this.dialogErrorSource.next(error.headers.get('X-artemisApp-error')!);
```

Comment:
This seems to be an uncommon thing to do. From what I see, it's usually done like this:
```suggestion
                this.dialogErrorSource.next(error.message);
```

Reply:
i found a way to actually trigger an error and found that what i wrote (and what existed for the delete and reset buttons, where i copied it from) doesn't work so i changed it as suggested and now in case of an error an alert with a (somewhat relevant) message is actually shown.